packages/realm/src/app-services/BaseSubscriptionSet.ts

```diff
@@ -202,11 +179,6 @@ export abstract class BaseSubscriptionSet {
     return subscription ? (new Subscription(subscription) as Subscription) : null; // TODO: Remove the type assertion into Subscription
   }
 
-  /**
-   * Enables index accessing when combined with the proxy handler's `get()` method.
```

Comment:
Creating a thread here to more easily discuss removing index accessing:

Previous comments:
> @kraenhansen
> It does make the tests a little more complicated to read / maintain.
>
> Perhaps they could benefit from some destructuring themselves, ex:
>
> `const [sub0, sub1, sub2] = subscriptions;`

> @RedBeard0531 
> Or use the findByName() API any time we need to be able to refer to a specific subscription.

Yeah I noticed particularly when refactoring the tests that the accessing by index felt more convenient. If having to use `findByName()`, users would have to always name their subscriptions.

Any more thoughts on this?

Reply:
Names are stable identifiers. Indexes are not. Users shouldn't have any expectation that the index<->subscription mapping is stable over time. If they need to refer to a subscription later, that is what names are for.

Reply:
True. Seems a bit difficult to know how convenient [0] would be for users.. Or how inconvenient not having it would be?

Reply:
I suspect it's mostly in the context of these tests that it gets complicated. Building an app I would probably store a reference to the subscription and use that to remove it later.

That being said, (and just to make it obvious) having en index accessor is not a prerequisite for the snippet I posted.